src/lightning/fabric/strategies/strategy.py

```diff
@@ -111,6 +112,17 @@ def process_dataloader(self, dataloader: DataLoader) -> DataLoader:
         """
         return dataloader
 
+    @contextmanager
+    def module_init_context(self) -> Generator:
```

Comment:
This is the default implementation, which sets device and dtype for new tensors.
Only in PyTorch 2.0 is it easy to change the default device like this. If support for torch < 2.0 is desired, we can hack something together.

Reply:
I don't think we need it. Should we emit a warning for `_TORCH_GREATER_EQUAL_2_0 == False`?

Reply:
Makes sense, added and documented!